src/basic/sysctl-util.c

```diff
@@ -44,8 +44,28 @@ char *sysctl_normalize(char *s) {
         return s;
 }
 
-int sysctl_write(const char *property, const char *value) {
+static int sysctl_f_ofd_lk(const char *p, bool set) {
+        struct flock flock = {
+                .l_type = F_WRLCK,
+                .l_whence = SEEK_SET,
+        };
+        _cleanup_close_ int fd = -EBADF;
+        int ret;
+
+        fd = open(p, O_RDWR);
+        if (fd < 0)
+                return -ENOENT;
+
+        ret = fcntl(fd, set ? F_OFD_SETLK : F_OFD_GETLK, &flock);
+        if (ret)
+                return -EBUSY;
+
+        return 0;
+}
+
+int sysctl_write(const char *property, const char *value, bool lock) {
         char *p;
+        int ret;
```

Comment:
all across our tree we store the return values of functions in a variable `int r;` (at least if it is a negative-errno style return value). please follow that naming here too.